include/openthread/coap.h

```diff
@@ -60,6 +60,10 @@ extern "C" {
 
 #define OT_COAP_MAX_TOKEN_LENGTH 8 ///< Max token length as specified (RFC 7252).
 
+#define OT_COAP_MAX_RETRANSMIT 30 ///< Max retransmit supported by OpenThread.
+
+#define OT_COAP_MIN_ACK_TIMEOUT 1000 ///< Minimal ACK timeout in milliseconds supported by OpenThread.
```

Comment:
@sjlongland, does a minimum ACK timeout of 1 second fit with your use case(s)?

Reply:
I think it should… I'd imagine you really don't want to be pummelling a low-rate mesh network at millisecond intervals with retransmissions.  I was thinking something in the realm of ~5 seconds or so.

A minimum of one would be fine.

Reply:
The 1 second value comes from CoAP RFC7252 4.8.1.

Reply:
I think someone might requires a smaller timeout for the first retransmission. We should allow extreme use cases when the user need it. 1000ms is too much larger than the minimal timeout that OpenThread can support theoretically. Thoughts?

Reply:
How probable is it that a response to a request is received within a sub-second interval?

On big and very busy mesh networks (30~50 nodes), a simple ICMP echo request takes ~2000ms to return.  A sub-second ACK time-out would have expired by then.

Reply:
I tend to leave this when we really hit a use case that requires sub-second ACK timeout. @simonlingoogle Thoughts?

Reply:
> The 1 second value comes from CoAP RFC7252 4.8.1.

I am fine with using 1000 ms as min timeout and changing to smaller timeout if we find it necessary in the future.